processor/intracesampler/murmur3.go

```diff
@@ -0,0 +1,69 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package intracesampler
+
+// hash is a murmur3 hash function, see http://en.wikipedia.org/wiki/MurmurHash
+func hash(key []byte, seed uint32) (hash uint32) {
```

Comment:
fair enough.
Can you please suggest the best way to do it in go? should I add a new module with the murmur3 code and consume it from both probabilistic sampler and this sampler?

Reply:
You would move the code to pkg/internal so both components can import it. It can be dine with a follow up PR, just file an issue for it.

Reply:
looks like the murmur3 hash was replaced with another hash [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/17837) after I opened the PR.
I will update the code accordingly

Reply:
Updated the PR with the new hash algorithm, CI is green. I think it will be best to extract the common code after this one merge so this PR will not make changes to other components.

Should I open the issue now or after we merge?